### ISSUE TITLE
CA2252 Preview Feature Analyzer not reporting diagnostics as error by default

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/roslyn-analyzers#5636]
### Analyzer

**Diagnostic ID**: [CA2252](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2252): `This API requires opting into preview features`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: 6.0.100-rtm.21514.17

### Describe the bug

Consuming preview APIs in the default console template is not producing build errors as expected. Instead, diagnostics are reported at Info level.

### Steps To Reproduce

1. `dotnet new console`
2. Update `Program.cs` to have the following single line of code (note that `System.Runtime.Intrinsics.X86.AvxVnni` is annotated as a preview feature)
3. `dotnet build`

```c#
Console.WriteLine(System.Runtime.Intrinsics.X86.AvxVnni.IsSupported);
```

### Expected behavior

A build error indicating that using `AvxVnni` requires opting into preview features.

### Actual behavior

* The build was successful with no warnings or errors.
* In a Visual Studio build, the following is reported:

| Severity | Code | Description | Project | File | Line | Suppression State | Detail Description |
| - | - | - | - | - | - | - | - |
| Message | CA2252 | Using 'IsSupported' requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information. | previewfeature | C:\Users\jeffhand\git\test\previewfeature\Program.cs | 1 | Active | An assembly has to opt into preview features before using them. |

## Additional context

In Visual Studio, the CA2252 analyzer's properties show with the following:

* Category: Usage
* Default severity: Error
* Effective severity: Info
* Enabled by default: True
* Tags: Telemetry EnabledRuleInAggressiveMode

This is potentially a blocking issue for 6.0 GA as we need preview features to be reported as build errors unless consumers opt in.

/cc @terrajobst @buyaa-n @tannergooding

**Comment by terrajobst** [comment]
> This is potentially a blocking issue for 6.0 GA as we need preview features to be reported as build errors unless consumers opt in.

For sure. I'd consider this a ship stopper.

**Comment by mavasani** [comment]
Confirmed this is due to the wrong entry in the above shipped releases file, which generates the following entry in `C:\Program Files\dotnet\sdk\6.0.100-rc.2.21505.57\Sdks\Microsoft.NET.Sdk\analyzers\build\config\AnalysisLevel_6_default.editorconfig`

```editorconfig
# CA2252: This API requires opting into preview features
dotnet_diagnostic.CA2252.severity = suggestion
```

**Comment by mavasani** [comment]
@jeffhandley You may also want to quickly check default severity of all entries for new .NET6 analyzers here matches your expectations: https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md#release-60

**Comment by jeffhandley** [comment]
> @jeffhandley You may also want to quickly check default severity of all entries for new .NET6 analyzers here matches your expectations: https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/AnalyzerReleases.Shipped.md#release-60

Good idea. @buyaa-n Can you do that please?

**Comment by buyaa-n** [comment]
> You may also want to quickly check default severity of all entries for new .NET6 analyzers here matches your expectations

For a few analyzers no info of expected severity or not discussed in the API review (though I have no issue with current severity). 

I don't see the CA1849, `UseAsyncMethodInAsyncContext` analyzer in the 6.0 shipped list (we are merged into 6.0), it is still in the unshipped list, not sure if that is an issue or it will be shipped later CC @jmarolf 

Other than that the list looks good to me
